framework-generators/messaging-adapter-generator/src/main/java/uk/gov/justice/subscription/jms/core/JmsUriToDestinationTypeConverter.java

```diff
@@ -0,0 +1,54 @@
+package uk.gov.justice.subscription.jms.core;
+
+import static org.apache.commons.lang3.StringUtils.trimToEmpty;
+import static uk.gov.justice.services.core.annotation.Component.EVENT_PROCESSOR;
+
+import java.util.Optional;
+
+import javax.jms.Destination;
+import javax.jms.Queue;
+import javax.jms.Topic;
+
+
+public class JmsUriToDestinationTypeConverter {
+
+    private static final String QUEUE = "queue";
+    private static final String TOPIC = "topic";
+
+    /**
+     * Retrieves the destination from jmsUri in the format jms:queue:some-more-string
+     *
+     * @return Optional of Queue.class, Topic.class or null
+     */
+    Optional<Class<? extends Destination>> convert(final String jmsUri) {
+
+        final String[] parts = trimToEmpty(jmsUri).split(":");
+        final String destinationType = (parts.length > 1) ? parts[1] : null;
+
+        if (QUEUE.equalsIgnoreCase(destinationType)) {
+            return Optional.of(Queue.class);
+        }
+
+        if (TOPIC.equalsIgnoreCase(destinationType)) {
+            return Optional.of(Topic.class);
+        }
+
+        return Optional.empty();
+    }
+
+    /**
+     * EVENT_PROCESSOR.
+     * For any other component, return null
+     *
+     * @param serviceComponent EVENT_PROCESSOR, EVENT_LISTENER, etc
+     * @param jmsUri           a fully blown jms URI
+     * @return a destination type by calling convert() only if the component type is an EVENT_PROCESSOR
+     */
+    public Optional<Class<? extends Destination>> convertForEventProcessor(final String serviceComponent, final String jmsUri) {
+        if (serviceComponent.contains(EVENT_PROCESSOR)) {
```

Comment:
Why are we using contains? (Is there a reason for this?)
If not using contains, it would be better to put the constant on the Left Hand Side to avoid NPE on serviceComponent

Reply:
Very good question @Morbu 
In fact, `equals` should just work.
I looked around and everywhere, contains was being used
https://github.com/CJSCommonPlatform/microservice-framework/blob/43f3839cb8e87b14a74c765ce773576c074804a2/framework-generators/messaging-adapter-generator/src/main/java/uk/gov/justice/subscription/jms/core/ComponentDestinationType.java#L50

Maybe there is a reason for it? I was not so sure.
I have added a null check here.

Thanks for the review